sympy/core/power.py

```diff
@@ -1686,16 +1686,19 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             e = logcombine(e).cancel()
 
         if not (m.is_zero or e.is_number and e.is_real):
-            res = exp(e*log(b))._eval_nseries(x, n=n, logx=logx, cdir=cdir)
-            if res is exp(e*log(b)):
-                return self
-            return res
+            if self == self._eval_as_leading_term(x, logx=logx, cdir=cdir):
+                res = exp(e*log(b))._eval_nseries(x, n=n, logx=logx, cdir=cdir)
+                if res == exp(e*log(b)):
+                    return self
+                return res
 
         f = b.as_leading_term(x, logx=logx)
         g = (b/f - S.One).cancel(expand=False)
         if not m.is_number:
             raise NotImplementedError()
         maxpow = n - m*e
+        if maxpow.has(Symbol):
+            maxpow += m*e
```

Comment:
Could this be `maxpow = n`?

Reply:
Yeah , I thought of doing that first but then `n` is a variable on type `int` and `m*e` is a variable of type `Mul` and eventually `n - m*e` would be a variable of type `Add` . So something like `mapow = n` won't work (probably something like importing sympify and using `mapow = sympify(n)` would be needed). Hence I thought of using this ,

The next line of line **1705** is the following which will return an `AttributeError` without n being sympified
```
        if maxpow.is_negative:
            return Order(x**(m*e), x)
```
I'll make the change as of now .